### PR TITLE
Add inbound queue

### PR DESCRIPTION
**Pull request by ParthDesai**

### CODE REVIEW

File: bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+use crate::{Error, Event, LOG_TARGET};
+use codec::DecodeAll;
+use core::marker::PhantomData;
+use frame_support::pallet_prelude::Weight;
+use snowbridge_core::Channel;
+use snowbridge_router_primitives::inbound::envelope::Envelope;
+use snowbridge_router_primitives::inbound::{MessageProcessor, VersionedXCMMessage};
+use sp_runtime::DispatchError;
+
+pub struct XCMMessageProcessor<T>(PhantomData<T>);
```
Comment (Agusrodri) [inline bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs:10]:
```suggestion
pub struct XcmMessageProcessor<T>(PhantomData<T>);
```

File: bridges/snowbridge/pallets/inbound-queue/src/lib.rs [hunk 2]
```diff
@@ -140,6 +138,8 @@ pub mod pallet {
 
 		/// To withdraw and deposit an asset.
 		type AssetTransactor: TransactAsset;
+
+		type MessageProcessor: MessageProcessor;
```
Comment (Agusrodri) [inline bridges/snowbridge/pallets/inbound-queue/src/lib.rs:142]: Maybe it's worth adding a short comment here explaining the purpose of the custom processor.

File: bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs (hunk 1 above)
Comment (ParthDesai) [reply bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs]: *fixed*

File: bridges/snowbridge/pallets/inbound-queue/src/lib.rs (hunk 2 above)
Comment (ParthDesai) [reply bridges/snowbridge/pallets/inbound-queue/src/lib.rs:143]: *fixed*

File: bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+use crate::{Error, Event, LOG_TARGET};
+use codec::DecodeAll;
+use core::marker::PhantomData;
+use snowbridge_core::Channel;
+use snowbridge_router_primitives::inbound::envelope::Envelope;
+use snowbridge_router_primitives::inbound::{MessageProcessor, VersionedXcmMessage};
+use sp_runtime::DispatchError;
+
+pub struct XcmMessageProcessor<T>(PhantomData<T>);
```
Comment (girazoki) [inline bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs:9]: I would probably call it some other thing but it is likely that snowbridge will tell you how to name it. So for me this is good enough

File: bridges/snowbridge/primitives/router/src/inbound/envelope.rs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+use snowbridge_core::{inbound::Log, ChannelId};
+
+use sp_core::{RuntimeDebug, H160, H256};
+use sp_std::prelude::*;
+
+use alloy_primitives::B256;
+use alloy_sol_types::{sol, SolEvent};
+
+sol! {
```
Comment (girazoki) [inline bridges/snowbridge/primitives/router/src/inbound/envelope.rs:11]: what is this macro?

**Review by girazoki — APPROVED** [review]
for me it looks good, but please can you investigate what `sol!` does?

File: bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs (hunk 3 above)
Comment (ParthDesai) [reply bridges/snowbridge/pallets/inbound-queue/src/xcm_message_processor.rs:9]: Reason why I named it `XcmMessageProcessor` as it expects incoming message to be converted to xcm.  :) But, let's see what snowbridge thinks.

File: bridges/snowbridge/primitives/router/src/inbound/envelope.rs (hunk 4 above)
Comment (ParthDesai) [reply bridges/snowbridge/primitives/router/src/inbound/envelope.rs:11]: `sol!`  macro takes an event declaration in solidity and provides helper functions and also struct to work with that event data. Like encoding/decoding etc.

**Pull request merged**